rpc/service/src/service.rs

```diff
@@ -620,6 +621,35 @@ NOTE: This error usually indicates an RPC conversion error between the node and
         Ok(GetDaaScoreTimestampEstimateResponse::new(timestamps))
     }
 
+    async fn get_utxo_return_address_call(&self, request: GetUtxoReturnAddressRequest) -> RpcResult<GetUtxoReturnAddressResponse> {
+        let session = self.consensus_manager.consensus().session().await;
+
+        let mut return_address = None;
+
+        // Convert a SPK to an Address
+        if let Some(spk) = session.async_get_utxo_return_script_public_key(request.txid, request.accepting_block_daa_score).await {
+            let script = spk.script();
+
+            // Standard Address scripts are only either 34 or 35 in length:
```

Comment:
in `kaspa_txscript` crate we have `ScriptClass::from_script()` which will return an enum `ScriptClass`:

```rs
pub enum ScriptClass {
    /// None of the recognized forms
    NonStandard = 0,
    /// Pay to pubkey
    PubKey,
    /// Pay to pubkey ECDSA
    PubKeyECDSA,
    /// Pay to script hash
    ScriptHash,
}
```

can this maybe be used to check if it is standard, by checking that it is not nonstandard?

Reply:
I did one better and refactored this to:

```
        // Convert a SPK to an Address
        let return_address = if let Some(spk) =
            session.async_get_utxo_return_script_public_key(request.txid, request.accepting_block_daa_score).await
        {
            extract_script_pub_key_address(&spk, self.config.prefix()).ok()
        } else {
            None
        };
```

In commit [56a4fcb](https://github.com/kaspanet/rusty-kaspa/pull/436/commits/56a4fcb07c9f4e2da42bde335f17e065b36c7a78)